android/app/src/main/java/app/organicmaps/car/screens/bookmarks/BookmarksLoader.java

```diff
@@ -0,0 +1,124 @@
+package app.organicmaps.car.screens.bookmarks;
+
+import android.graphics.drawable.Drawable;
+import android.location.Location;
+import android.text.SpannableStringBuilder;
+import android.text.Spanned;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.car.app.CarContext;
+import androidx.car.app.constraints.ConstraintManager;
+import androidx.car.app.model.CarIcon;
+import androidx.car.app.model.DistanceSpan;
+import androidx.car.app.model.ForegroundCarColorSpan;
+import androidx.car.app.model.ItemList;
+import androidx.car.app.model.Row;
+import androidx.core.graphics.drawable.IconCompat;
+
+import app.organicmaps.R;
+import app.organicmaps.bookmarks.data.BookmarkCategory;
+import app.organicmaps.bookmarks.data.BookmarkInfo;
+import app.organicmaps.bookmarks.data.BookmarkManager;
+import app.organicmaps.bookmarks.data.Icon;
+import app.organicmaps.car.util.Colors;
+import app.organicmaps.car.util.RoutingHelpers;
+import app.organicmaps.location.LocationHelper;
+import app.organicmaps.util.Distance;
+import app.organicmaps.util.Graphics;
+import app.organicmaps.util.concurrency.ThreadPool;
+import app.organicmaps.util.concurrency.UiThread;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class BookmarksLoader
+{
+  public interface OnBookmarksLoaded
+  {
+    void onBookmarksLoaded(@NonNull ItemList bookmarks);
+  }
+
+  // The maximum size should be equal to ConstraintManager.CONTENT_LIMIT_TYPE_LIST.
+  // However, having more than 50 items results in android.os.TransactionTooLargeException.
```

Comment:
Is there a search that allows to find a bookmark despite the limit of 50?

Reply:
I'm not following you. Could you please clarify the connection between this screen and the search functionality?

Reply:
1. If there is a limit, user can't access >50 bookmarks.
2. If there is a bookmarks search somewhere, user can access >50 bookmarks.

Reply:
1. Yes
2. Yes

This screen has nothing related to the search.

Reply:
The relation is like this: if users can't find 55th bookmark in this list, at least they can use the search from another screen to find that 55th bookmark. If the search is implemented, of course.

If there is no search, users can't reach 55th bookmark in any way.